### ISSUE TITLE
missing nodes/tools (dynamo 2.0)

### ISSUE THREAD

**Issue by erfajo** [DynamoDS/Dynamo#8547]
## Dynamo version
DynamoRevit2.0.0.20180206T1942

## Operating system
Windows 10 enterprise x64

## What did you do? 
Started dynamo...

## What did you expect to see?
I expected that nodes/tools were available... but there is nothing at all, it is absolutely empty!

Last version that worked was DynamoRevit2.0.0.20180111T0104, but it is not possible to install that version anymore, I have tried. So please fix whatever error the latest version has.

I get seven errors, don't know if that has any influence since I am used to having a number of errors also in the 1.3.2 version and that works fine.
![image](https://user-images.githubusercontent.com/2844374/36015949-08b3d02e-0d71-11e8-9f92-c8160bb37286.png)

## What did you see instead?
Closed dynamo...

**Comment by erfajo** [comment]
If I start the sandbox version, then it looks like expected!?

Is there a migration problem from earliere versions?

![image](https://user-images.githubusercontent.com/2844374/36016657-602a9768-0d74-11e8-9b81-0a5655d997ee.png)

**Comment by ramramps** [comment]
@erfajo  what is the version of Revit ?

**Comment by andydandy74** [comment]
I'm not even seeing a library in the sandbox (same build as @erfajo ).
![grafik](https://user-images.githubusercontent.com/3014437/36031094-ee782276-0da9-11e8-879f-d5182bdd52d3.png)

**Comment by kronz** [comment]
@ramramps @mjkkirschner Note that Andreas is seeing this in Sandbox, that is, not in conflict with 2018.2's CEF sharp.

**Comment by andydandy74** [comment]
@ramramps @mjkkirschner FYI This is in a VM

**Comment by ramramps** [comment]
@andydandy74 I got this build (sandbox) and I checked it on my machine. It looks ok. 
Can you get us the VM configuration?

**Comment by andydandy74** [comment]
@ramramps You mean the *vmxf file?

**Comment by ramramps** [comment]
Like the windows version, Graphics card...

**Comment by mjkkirschner** [comment]
@andydandy74 were you seeing the library in your VM on previous versions of 2.x?

**Comment by mjkkirschner** [comment]
@erfajo please try removing the pNID modeler addin folder from the addins folder under your revit install directory.

**Comment by jnealb** [comment]
@erfajo @ramramps @andydandy74 @kronz Installing DynamoRevit2.0.0.20180206T1942 on a Win10 VM with Revit 2018.3 pre-installed gives correct results with both D4R and Sandbox. Please note that the running Core version is 2.0.0.4006 as reported by the About box. Is it possible that this might be noise from an incomplete "intra 2.0" uninstall/install sequence?

**Comment by andydandy74** [comment]
@ramramps
VMware Workstation 
Graphics card: VMware SVGA 3D
System stats in VM:
![grafik](https://user-images.githubusercontent.com/3014437/36035395-01c64508-0db7-11e8-82d8-cb446f22c589.png)

@mjkkirschner 
This is the first install of 2.0 on this VM. So far I've only had it installed on my home machine.
I will uninstall this build, install an older build to check the library visibility and report back.

**Comment by jnealb** [comment]
ditto on the gfx subsystem: 
![image](https://user-images.githubusercontent.com/9357222/36035734-c6c75b4e-0d85-11e8-8b04-486db5ff98cb.png)

**Comment by jnealb** [comment]
Because I have installed 2.0 before on this VM my test is not quite fair. Should do it on a freshly minted VM.

**Comment by andydandy74** [comment]
FWIW I have the same issue on my Win 10 VM :-(

**Comment by jnealb** [comment]
I have a pristine Lenovo with Win10. Will try there.

**Comment by mjkkirschner** [comment]
@andydandy74 do you see anything library related in the console output?

**Comment by andydandy74** [comment]
Tried the first daily build of both January and December, same result in Revit 2017.2.2 (no sandbox available yet in those builds).

**Comment by mjkkirschner** [comment]
also @andydandy74 just to be sure - have you enabled 3d acceleration in your VM?

**Comment by andydandy74** [comment]
@mjkkirschner I do. Rough translation:
```
LibraryUI: The file or assembly "CefSharp.Core.dll" or a dependency thereof was not found. The module could not be found.
Workspace is being emptied.
```

**Comment by andydandy74** [comment]
@mjkkirschner yes, 3d acceleration is enabled :-)

**Comment by ramramps** [comment]
can you check the install location and see whether the file is there?
<img width="870" alt="screen shot 2018-02-09 at 11 09 00 am" src="https://user-images.githubusercontent.com/8495197/36037112-b85f3370-0d89-11e8-83cc-343b089e1c13.png">

**Comment by andydandy74** [comment]
@ramramps 
![grafik](https://user-images.githubusercontent.com/3014437/36037224-6211570e-0dbc-11e8-80c6-c76ec42affbb.png)

**Comment by mjkkirschner** [comment]
https://github.com/cefsharp/CefSharp/wiki/Frequently-asked-questions#4-why-do-i-get-an-error-about-could-not-load-file-or-assembly-cefsharpcoredll-or-one-of-its-dependencies-the-specified-module-could-not-be-found-when-trying-to-run-my-cefsharp-based-application-it-compiles-successfully-but-does-not-run-it-runs-on-my-developer-machine-though-throws-an-exception-when-i-copy-it-to-another-computer


worth checking what version of vc++ distrub is installed - or if there are missing dlls relative to a build of dynamo master branch... my guess is that it's likely a missing installed dependency like .net framework version or vc++ version which we have neglected to install.

since @erfajo 's install works, I think we are probably correctly delivering all CEF dlls, but not its deps.

**Comment by jnealb** [comment]
@mjkkirschner @andydandy74 @alfarok What exactly should I install on my pristine Lenovo? Revit 2017.2.2 first?

**Comment by andydandy74** [comment]
@mjkkirschner 
these are the vc++ redistributables installed on the VM
![grafik](https://user-images.githubusercontent.com/3014437/36037650-af64d11a-0dbd-11e8-866d-d0b51d7ccfb2.png)
As for .NET, there are 4.6.00081 and 4.7.02053

**Comment by ramramps** [comment]
<img width="1203" alt="screen shot 2018-02-09 at 11 28 31 am" src="https://user-images.githubusercontent.com/8495197/36038022-6f61fd94-0d8c-11e8-8875-df449957d71a.png">

**Comment by ramramps** [comment]
I think you need VC++ 2013 distributable.

**Comment by jnealb** [comment]
This is a known issue and tracked with DYN-578.

**Comment by erfajo** [comment]
Sorry for my absence yesterday, I had other matters to address yesterday.

I am glad to see that the issue got some attention, but the suggested solution that VC++ 2013 should be the problem is NOT the case. It is the PnIDModeler there is the problem. After removing that Addin my dynamo session worked as expected.

BUT, the PnIDModeler is an Autodesk plugin, that comes with Revit 2018. I don’t know which service that uses PnIDModeler, but I have an idea that PnIDModeler and Fabrication Parts has some kind of connection.

When Revit is being installed the Fabrication Parts is being suggested (via Autodesk desktop app) as an updated. I accepted that update. Have you tried to update Fabrication to the latest version to see if that gave errors with Dynamo 2.0? You must have made some kind of change since you end up with errors in this combination of Dynamo and PnIDModeler installed.

PnIDModeler
https://knowledge.autodesk.com/support/revit-products/learn-explore/caas/CloudHelp/cloudhelp/2018/ENU/Revit-Model/files/GUID-0DE5B102-A0B5-4B3C-AF59-1676DAB997C2-htm.html

Fabrication Parts
https://knowledge.autodesk.com/support/revit-products/learn-explore/caas/CloudHelp/cloudhelp/2016/ENU/Revit-Model/files/GUID-436D69AD-D2A5-46DA-BB16-0450F29091B4-htm.html

**Comment by ramramps** [comment]
@erfajo Thanks. VC++ 2013 is the issue for @andydandy74. Different issues got crumbled here. 
It's the CEF version issue. Revit 2018.2 has Cef version different from Dynamo. And CEF in Revit 2018.2 is inside PnIDModeler. So, when you deleted the PnIDModeler, Dynamo loads its CEF version in the process and it worked as expected. This is the issue only in Revit 2018.2 and this is fixed in Revit 2018.3 or Revit 2019.

**Comment by erfajo** [comment]
Thanks, that sounds good. I can live with testing Dynamo 2.0 by moving PnIDModeler out of the addin folder. If I should end having use of the PnIDModeler I can move it back again.
